internal/cli/atlas/setup/setup_cmd.go

```diff
@@ -48,14 +108,281 @@ type ProfileReader interface {
 type Opts struct {
 	cli.GlobalOpts
 	cli.WatchOpts
-	quickstart quickstart.Flow
-	register   auth.RegisterOpts
-	config     ProfileReader
+	register            auth.RegisterOpts
+	config              ProfileReader
+	store               store.AtlasClusterQuickStarter
+	LabelKey            string
+	LabelValue          string
+	defaultName         string
+	ClusterName         string
+	Provider            string
+	Region              string
+	IPAddresses         []string
+	IPAddressesResponse string
+	DBUsername          string
+	DBUserPassword      string
+	SampleDataJobID     string
+	Tier                string
+	Tag                 map[string]string
+
+	SkipSampleData              bool
+	SkipMongosh                 bool
+	DefaultValue                bool
+	Confirm                     bool
+	CurrentIP                   bool
+	EnableTerminationProtection bool
+
+	flags   *pflag.FlagSet
+	flagSet map[string]struct{}
+
 	// control
 	skipRegister bool
 	skipLogin    bool
 }
 
+type setup struct {
```

Comment:
```suggestion
type clusterSettings struct {
```
[nit] I know this was migrated from quickstart (where it was called `quickstart`) but might be good to have a name that reflects the purpose of this struct